padd.sh

```diff
@@ -1054,7 +1041,7 @@ json_extract() {
     local pair_regex="\"${key}\"[[:space:]]*:[[:space:]]*(${value_regex})"
 
     if [[ ${json} =~ ${pair_regex} ]]; then
-        echo $(sed 's/^"\|"$//g' <<< "${BASH_REMATCH[1]}")
+        echo "${BASH_REMATCH[1]//^"\|"$/}"
```

Comment:
This does not work.
The `sed` removed the leading and trailing `"`. If you want to use bash's built-in pattern substitution you need different ancors (https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion).

I would stick with sed but make it a bit more obvious what happens here
```suggestion
        # remove leading and trailing quotes
        sed -e 's/^"//' -e 's/"$//' <<<"${BASH_REMATCH[1]}"
```

Reply:
Ups, I didn't test this json output… 😱 

But in both my outputs (old/new) are always leading and trailing quotes - regardless if using shell-parameter-expansion or `sed`… 🤔 (with alias pihole='docker exec -it pihole $*')
```
🦎🖥  ✔ ~/temp/PRs/PADD [issue_188_fix_SC2034_and_SC2154|✔]
15:04 $ pihole /etc/pihole/padd.sh.issue_188_fix_SC2034_and_SC2154 -j
{"domains_being_blocked":4595712,"dns_queries_today":49081,"ads_blocked_today":8439,"ads_percentage_today":17.194027,"clients": 17}

🦎🖥  ✔ ~/temp/PRs/PADD [issue_188_fix_SC2034_and_SC2154|✔]
15:05 $ pihole /etc/pihole/padd.sh -j
{"domains_being_blocked":4595712,"dns_queries_today":49082,"ads_blocked_today":8439,"ads_percentage_today":17.193676,"clients": 17}

```

Reply:
This will remove all quotes, not only the leading and trailing

Reply:
Ok, I'm a little bit confused now 😕 
I have a nightly cronjob that's writing daily stats of `padd.sh`'s json-output in a logfile:
> echo -e $(date +\%Y-\%m-\%d) $(docker exec --tty pihole /etc/pihole/padd.sh -j) >> ~/logs/pihole-padd.log

Which results in this:
```
…
2022-03-20 {"domains_being_blocked":4390857,"dns_queries_today":60925,"ads_blocked_today":10889,"ads_percentage_today":17.872795,"clients": 20}
2022-03-21 {"domains_being_blocked":4390857,"dns_queries_today":45849,"ads_blocked_today":8844,"ads_percentage_today":19.289406,"clients": 19}
2022-03-22 {"domains_being_blocked":4595712,"dns_queries_today":48666,"ads_blocked_today":10259,"ads_percentage_today":21.080425,"clients": 19}
2022-03-23 {"domains_being_blocked":4595712,"dns_queries_today":50859,"ads_blocked_today":9366,"ads_percentage_today":18.415619,"clients": 20}
```

Now, which quotes do you want to have to be removed which my `shellcheck`ed version doesn't? 🤔

As I wrote [before](https://github.com/pi-hole/PADD/pull/206#discussion_r834349744) the json-output of old and new `padd.sh` looks very much the same for me 🤷🏻‍♂️

Reply:
Sorry, I was mislead by the "json" in function name…

Line 13 is wrong (with quotes), line 14 and line 15 are both good (both without quotes) in my eyes. `shellcheck` is fine with both, but you prefered the `sed` solution. Now that I've understood, I totally agree with your suggestion 👍🏻 
```
🦎🖥  ✔ ~/temp/PRs/PADD [issue_188_fix_SC2034_and_SC2154|✚ 1…1]
22:35 $ bat 1
───────┬──────────────────────────────────────────────────────────────────────────────────────
       │ File: 1
───────┼──────────────────────────────────────────────────────────────────────────────────────
   1   │ #!/bin/bash
   2   │
   3   │ json_extract() {
   4   │     local key=$1
   5   │     local json=$2
   6   │
   7   │     local string_regex='"([^"\]|\\.)*"'
   8   │     local number_regex='-?(0|[1-9][0-9]*)(\.[0-9]+)?([eE][+-]?[0-9]+)?'
   9   │     local value_regex="${string_regex}|${number_regex}|true|false|null"
  10   │     local pair_regex="\"${key}\"[[:space:]]*:[[:space:]]*(${value_regex})"
  11   │
  12   │     if [[ ${json} =~ ${pair_regex} ]]; then
  13   │         echo $LINENO "${BASH_REMATCH[1]//^"\|"$/}"
  14   │         echo $LINENO "${BASH_REMATCH[1]//\"/}"
  15   │         echo $LINENO $(sed -e 's/^"//' -e 's/"$//' <<<"${BASH_REMATCH[1]}")
  16   │     else
  17   │         return 1
  18   │     fi
  19   │ }
  20   │
  21   │
  22   │ padd_version_latest=$(json_extract tag_name "$(curl -s 'https://api.github.com/repos/pi-hole/PADD/releases/latest' 2> /dev/null)")
  23   │
  24   │ echo ${padd_version_latest}
───────┴──────────────────────────────────────────────────────────────────────────────────────
🦎🖥  ✔ ~/temp/PRs/PADD [issue_188_fix_SC2034_and_SC2154|✚ 1…1]
22:35 $ bash 1
13 "v3.6.6" 14 v3.6.6 15 v3.6.6
```